api/agent/agent.go

```diff
@@ -768,9 +768,9 @@ InitPhase:
 			return
 		case <-a.shutWg.Closer(): // agent shutdown
 			return
-		case <-callerCtx.Done(): // original caller disconnected or serviced by another container?
+		case <-caller.done: // original caller disconnected or serviced by another container?
 			evictor.SetEvictable(true)
-			callerCtx = context.Background() // block 'callerCtx' after this point
+			caller.done = make(chan struct{}) // block 'doneChan' after this point
```

Comment:
nil would work and not allocate as well, not that this is a very hot path, just for future reference (it's not widely known that receive on nil channel doesn't panic and can be useful sometimes, like here)